### DISCUSSION TITLE
translate cmdline to Magick.net (distort polar)

### DISCUSSION THREAD

**Discussion by aquiel2k** [dlemstra/Magick.NET#669] [Help] [answered]
I use the following cmdline to get a ring from a line (similar to this [https://www.imagemagick.org/Usage/distorts/#polar](url) 

`convert.exe "line.png" -virtual-pixel HorizontalTile -background White +distort Polar "340,145 0,0 0,360"  "ring.png"`

if i use this
```
Dim Magick As New MagickImage()
Magick.Read("line.png")
Magick.VirtualPixelMethod = VirtualPixelMethod.HorizontalTile
Magick.BackgroundColor = MagickColors.White
Magick.Distort(DistortMethod.Polar, {340, 145, 0, 0, 0, 360})
Magick.Write("ring.png")
```
the result ist not a ring but a quarter circle
I tried to find more documentaion how to transtlate the cmdline but did not find anything 
I would be very grateful for your help.

**Comment by dlemstra** [comment, marked as answer]
You are very close. You will need to add `DistortSettings` to the `Distort` method and set the `Bestfit` to `true`. The `+distort` option means that `Bestfit` is set to `true`. And with `-distort` this option is set to `false`.

p.s. The `MagickImage` class is `IDisposable` so you should either use `using` or `Dispose()` the image.

**Comment by aquiel2k** [comment]
Thank you. It works now.
